app/controllers/correctReturn/CorrectReturnCheckChangesCYAController.scala

```diff
@@ -19,30 +19,39 @@ package controllers.correctReturn
 import com.google.inject.Inject
 import config.FrontendAppConfig
 import controllers.actions.ControllerActions
+import models.SdilReturn
 import models.SelectChange.CorrectReturn
+import models.correctReturn.ChangedPage
+import orchestrators.CorrectReturnOrchestrator
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import views.html.correctReturn.CorrectReturnCheckChangesCYAView
 import views.summary.correctReturn.CorrectReturnBaseCYASummary
 
+import scala.concurrent.ExecutionContext
+
 class CorrectReturnCheckChangesCYAController @Inject()(
                                             override val messagesApi: MessagesApi,
                                             controllerActions: ControllerActions,
                                             val controllerComponents: MessagesControllerComponents,
+                                            val correctReturnOrchestrator: CorrectReturnOrchestrator,
                                             view: CorrectReturnCheckChangesCYAView
-                                          )(implicit config: FrontendAppConfig) extends FrontendBaseController with I18nSupport {
+                                          )(implicit config: FrontendAppConfig, ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
   def onPageLoad(): Action[AnyContent] = controllerActions.withCorrectReturnJourneyData {
     implicit request =>
-
       val orgName: String = " " + request.subscription.orgName
-      val sections = CorrectReturnBaseCYASummary.summaryListAndHeadings(request.userAnswers, request.subscription)
+      val originalSDILReturn = request.userAnswers.getCorrectReturnOriginalSDILReturnData.get
```

Comment:
Any suggestions for a safer way to obtain this is greatly appreciated.  It should be in the user answers, and once guards are in place the user should not reach this page unless everything is in user answers, but as is, if the sdil return is not in user answers we get a 500 - it does not feel right to leave it an open get.  A get or else makes it optional, which would also require a change on ChangedPages model.  
Best practice is to?

Reply:
Use a `.map` on `request.userAnswers.getCorrectReturnOriginalSDILReturnData` and `Redirect` to somewhere sensible if it is not there. That's probably the simplest, least bad solution. But not the thorough solution that is probably hinted at by the code - perhaps a further improvement though.